[MI-2814] Done the review fixes of github PR #636

Nityanand13 opened the pull request:
PR [#636](https://github.com/mattermost/mattermost-plugin-github/pull/636)

Nityanand13 committed:
[MI-2736]: Review fixes done
1. Improved code readability

Nityanand13 committed:
Merge branch 'master' of github.com:mattermost/mattermost-plugin-github into MI-2736

Nityanand13 committed:
[MI-2736]: Review fixes done
1. Fixed linting errors

Nityanand13 committed:
Merge branch 'MM-618' of github.com:Brightscout/mattermost-plugin-github into MI-2736

Nityanand13 committed:
[MI-2736]: Review fixes done
1. Fixed linting error

Nityanand13 committed:
[MI-2736]: Review fixes done
1. Improved code readability

Nityanand13 committed:
Merge branch 'MM-618' of github.com:Brightscout/mattermost-plugin-github into MI-2736

Nityanand13 committed:
[MI-2736]: Review fixes done
1. Changed the case of few endpoints to snake case

Nityanand13 committed:
[MI-2814]: Review fixes done of github PR #636

Nityanand13 committed:
Merge branch 'MM-618' of github.com:Brightscout/mattermost-plugin-github into MI-2814

Kshitij-Katiyar commented on webapp/src/components/modals/create_update_issue/create_update_issue.jsx:
```diff
@@ -46,36 +48,53 @@ export default class CreateOrUpdateIssueModal extends PureComponent {
+            labels = [];
+        }
+
+        this.setState({assignees});
```
i think you can set all the states in a single line

Kshitij-Katiyar commented on webapp/src/components/modals/create_update_issue/create_update_issue.jsx:
```diff
@@ -46,36 +48,53 @@ export default class CreateOrUpdateIssueModal extends PureComponent {
+    }
+
+    updateState(issueInfo) {
+        var {channel_id, title, description, assignees, labels, milestone_title, milestone_number, repo_full_name} = issueInfo ?? {};
```
are you updating any of these variable if not use const

manojmalik20 replied:
Use `const`.

Kshitij-Katiyar commented on webapp/src/components/modals/create_update_issue/create_update_issue.jsx:
```diff
@@ -46,36 +48,53 @@ export default class CreateOrUpdateIssueModal extends PureComponent {
+    }
+
+    updateState(issueInfo) {
+        var {channel_id, title, description, assignees, labels, milestone_title, milestone_number, repo_full_name} = issueInfo ?? {};
```
any reason that for issueInfo you have used snake case but casing is different on line 57

Nityanand13 replied:
Discussed

Nityanand13 committed:
[MI-2814]: Review fixes done
1. Made comment field editable when trying to add a comment from message action.

Kshitij-Katiyar commented on webapp/src/components/modals/attach_comment_to_issue/attach_comment_to_issue.jsx:
```diff
@@ -106,9 +106,17 @@ export default class AttachIssueModal extends PureComponent {
         });
     };
 
+    /* eslint-disable react/no-did-update-set-state*/
```
use next or same line disable lint command use it for 112 line

Kshitij-Katiyar commented on webapp/src/components/modals/create_update_issue/create_update_issue.jsx:
```diff
@@ -252,6 +252,7 @@ export default class CreateOrUpdateIssueModal extends PureComponent {
                     onChange={this.handleIssueTitleChange}
                 />
                 {issueTitleValidationError}
+
```
why this change ?

Nityanand13 replied:
For consistency

Nityanand13 committed:
[MI-2814]: Review fixes done
1. Improved code readability

manojmalik20 commented on server/plugin/api.go:
```diff
@@ -97,9 +97,7 @@ func (p *Plugin) initializeAPI() {
-	apiRouter.HandleFunc("/edit_issue_modal", p.checkAuth(p.attachUserContext(p.openIssueEditModal), ResponseTypePlain)).Methods(http.MethodPost)
-	apiRouter.HandleFunc("/close_reopen_issue_modal", p.checkAuth(p.attachUserContext(p.openCloseOrReopenIssueModal), ResponseTypePlain)).Methods(http.MethodPost)
-	apiRouter.HandleFunc("/attach_comment_issue_modal", p.checkAuth(p.attachUserContext(p.openAttachCommentIssueModal), ResponseTypePlain)).Methods(http.MethodPost)
+	apiRouter.HandleFunc("/issue_info", p.checkAuth(p.attachUserContext(p.getIssueInfo), ResponseTypePlain)).Methods(http.MethodPost)
```
Why is the method `POST` when we are getting the issue info?

manojmalik20 commented on webapp/src/actions/index.js:
```diff
@@ -211,47 +211,11 @@ export function getMilestoneOptions(repo) {
 }
 
-export function attachCommentIssueModal(payload) {
+export function issueInfo(payload) {
```
```suggestion
export function getIssueInfo(payload) {
```

manojmalik20 commented on webapp/src/client/client.js:
```diff
@@ -7,16 +7,8 @@ import {ClientError} from 'mattermost-redux/client/client4';
-
-    attachCommentIssueModal = async (payload) => {
-        return this.doPost(`${this.url}/attach_comment_issue_modal`, payload);
+    issueInfo = async (payload) => {
```
```suggestion
    getIssueInfo = async (payload) => {
```

manojmalik20 commented on webapp/src/components/modals/create_update_issue/create_update_issue.jsx:
```diff
@@ -46,36 +48,53 @@ export default class CreateOrUpdateIssueModal extends PureComponent {
+
+    updateState(issueInfo) {
+        var {channel_id, title, description, assignees, labels, milestone_title, milestone_number, repo_full_name} = issueInfo ?? {};
+        if (!assignees) {
+            assignees = [];
+        }
```
```suggestion
        const assignees = issueInfo?.assignees ?? [];
```

manojmalik20 commented on webapp/src/components/modals/create_update_issue/create_update_issue.jsx:
```diff
@@ -46,36 +48,53 @@ export default class CreateOrUpdateIssueModal extends PureComponent {
-            },
-            issueDescription: description,
-            issueTitle: title.substring(0, MAX_TITLE_LENGTH)});
+        if (this.props.messageData && this.props.messageData.repo_owner && !prevProps.visible && this.props.visible) {
```
```suggestion
        if (this.props.messageData?.repo_owner && !prevProps.visible && this.props.visible) {
```

Nityanand13 committed:
[MI-2814]: Review fixes done
1. Used GET method on getting the issue info
2. Improved code readability

Nityanand13 committed:
[MI-2814]: Improved linting error
